src/ImageSharp/Processing/ProjectiveTransformBuilder.cs

```diff
@@ -0,0 +1,232 @@
+        public ProjectiveTransformBuilder(Rectangle sourceRectangle)
+        {
+            Guard.MustBeGreaterThan(sourceRectangle.Width, 0, nameof(sourceRectangle));
+            Guard.MustBeGreaterThan(sourceRectangle.Height, 0, nameof(sourceRectangle));
+
+            this.sourceRectangle = sourceRectangle;
+        }
+
+        /// <summary>
+        /// Gets the source image size.
+        /// </summary>
+        internal Size Size => this.sourceRectangle.Size;
+
+        /// <summary>
+        /// Prepends a matrix that performs a tapering projective transform.
+        /// </summary>
+        /// <param name="side">An enumeration that indicates the side of the rectangle that tapers.</param>
+        /// <param name="corner">An enumeration that indicates on which corners to taper the rectangle.</param>
+        /// <param name="fraction">The amount to taper.</param>
+        /// <returns>The <see cref="ProjectiveTransformBuilder"/>.</returns>
+        public ProjectiveTransformBuilder PrependTaperMatrix(TaperSide side, TaperCorner corner, float fraction)
+            => this.PrependMatrix(TransformUtils.CreateTaperMatrix(this.Size, side, corner, fraction));
+
+        /// <summary>
+        /// Appends a matrix that performs a tapering projective transform.
+        /// </summary>
+        /// <param name="side">An enumeration that indicates the side of the rectangle that tapers.</param>
+        /// <param name="corner">An enumeration that indicates on which corners to taper the rectangle.</param>
+        /// <param name="fraction">The amount to taper.</param>
+        /// <returns>The <see cref="ProjectiveTransformBuilder"/>.</returns>
+        public ProjectiveTransformBuilder AppendTaperMatrix(TaperSide side, TaperCorner corner, float fraction)
+            => this.AppendMatrix(TransformUtils.CreateTaperMatrix(this.Size, side, corner, fraction));
+
+        /// <summary>
+        /// Prepends a centered rotation matrix using the given rotation in radians.
+        /// </summary>
+        /// <param name="radians">The amount of rotation, in radians.</param>
+        /// <returns>The <see cref="ProjectiveTransformBuilder"/>.</returns>
+        public ProjectiveTransformBuilder PrependCenteredRotationRadians(float radians)
+        {
+            var m = new Matrix4x4(TransformUtils.CreateRotationMatrixRadians(radians, this.Size));
+            return this.PrependMatrix(m);
+        }
+
+        /// <summary>
+        /// Appends a centered rotation matrix using the given rotation in radians.
+        /// </summary>
+        /// <param name="radians">The amount of rotation, in radians.</param>
+        /// <returns>The <see cref="ProjectiveTransformBuilder"/>.</returns>
+        public ProjectiveTransformBuilder AppendCenteredRotationRadians(float radians)
+        {
+            var m = new Matrix4x4(TransformUtils.CreateRotationMatrixRadians(radians, this.Size));
```

Comment:
Well this is a pleasant surprise! I didn't know it had a constructor that too a 3x2 matrix :smile: